docs/topics/Gateway.md

```diff
@@ -177,6 +177,110 @@ If successful, the gateway will respond by replaying all missed events in order,
 
 If the gateway ever issues a disconnect to your client, it will provide a close event code that you can use to properly handle the disconnection. A full list of these close codes can be found in the [Response Codes](#DOCS_TOPICS_OPCODES_AND_STATUS_CODES/gateway-close-event-codes) documentation.
 
+## Gateway Intents
+
+> info
+> Intents are optionally supported on the v6 gateway. They will become mandatory in a future v7 gateway version.
+
+Maintaining a stateful application can be difficult when it comes to the amount of data you're expected to process, especially at scale. Gateway Intents are a system to help you lower that computational burden.
+
+When [identifying](#DOCS_TOPICS_GATEWAY/identifying) to the gateway, you can specify an `intents` parameter which allows you to conditionally subscribe to pre-defined "intents", groups of events defined by Discord. If you do not specify a certain intent, you will not receive any of the gateway events that are batched into that group. The valid intents are:
+
+### List of Intents
+
+```
+GUILDS (1 << 0)
+  - GUILD_CREATE
+  - GUILD_DELETE
+  - GUILD_ROLE_CREATE
+  - GUILD_ROLE_UPDATE
+  - GUILD_ROLE_DELETE
+  - CHANNEL_CREATE
+  - CHANNEL_UPDATE
+  - CHANNEL_DELETE
+  - CHANNEL_PINS_UPDATE
+
+GUILD MEMBERS (1 << 1)
+  - GUILD_MEMBER_ADD
+  - GUILD_MEMBER_UPDATE
+  - GUILD_MEMBER_REMOVE
+
+GUILD_BANS (1 << 2)
+  - GUILD_BAN_ADD
+  - GUILD_BAN_REMOVE
+
+GUILD_EMOJIS (1 << 3)
+  - GUILD_EMOJIS_UPDATE
+
+GUILD_INTEGRATIONS (1 << 4)
+  - GUILD_INTEGRATIONS_UPDATE
+
+GUILD_WEBHOOKS (1 << 5)
+  - WEBHOOKS_UPDATE
+
+GUILD_INVITES (1 << 6)
+  - INVITE_CREATE
+  - INVITE_DELETE
+
+GUILD_VOICE_STATES (1 << 7)
+  - VOICE_STATE_UPDATE
+
+GUILD_PRESENCES (1 << 8)
+  - PRESENCE_UPDATE
```

Comment:
So as weird as this might sound, I've got to ask: Does this intent cover *all* presence updates, or only those generated by guilds?

I ask because, lower down in the list, some intents make a distinction between where the event comes from (i.e. GUILD_MESSAGE_TYPING and DIRECT_MESSAGE_TYPING with the TYPING_START event), and I was wondering if the same distinction should apply here, as I seem to remember that PRESENCE_UPDATE could come in without an associated guild (which I think meant referring to the user overall rather than on one guild).

Reply:
As far as I know—and @night or @jhgg please correct me—all `PRESENCE_UPDATES` will have a corresponding `guild_id`. Bots today do not receive presence from open DMs.

Reply:
Correct yes.

Reply:
Well, that answers that question. Thanks!

Feel free to mark this as resolved. :)